test/test_nn.py

```diff
@@ -10414,6 +10401,47 @@ def test_multi_margin_loss_errors(self, device):
                           lambda: nn.functional.multi_margin_loss(torch.randn(5, device=device),
                                                                   torch.zeros(3, device=device)))
 
+    def test_nll_loss_mismatched_batch(self, device):
+        x = torch.randn((10, 3), requires_grad=True, device=device)
+        # t should have size (10,)
+        t = torch.zeros((3,), dtype=torch.int64, device=device)
+        with self.assertRaisesRegex(ValueError, 'Expected.*batch_size'):
+            F.nll_loss(x, t)
+
+    def test_nll_loss_out_of_bounds_ignore_index(self, device):
+        x = torch.randn(6, 3, requires_grad=True, device=device)
+        t = torch.tensor([0, 1, 255, 0, 1, 2], dtype=torch.int64, device=device)
+        for reduction in ['mean', 'none']:
+            F.nll_loss(x, t, ignore_index=255, reduction=reduction).sum().backward()
+
+    def test_nll_loss(self, device):
+        C = 10
+        for N in [0, 10]:
+            x = torch.rand(N, C, requires_grad=True, device=device)
+            target = torch.randint(C, (N,), device=device)
+            for reduction in ['none', 'mean', 'sum']:
+                y = F.nll_loss(x, target, reduction=reduction)
+                if reduction == 'none':
+                    self.assertTrue(y.size() == target.size())
+                else:
+                    self.assertEqual(y.dim(), 0)
+                y.sum().backward()
```

Comment:
We should check that `y.grad.size() == y.size()` to complete our check that the backwards pass handles the 0-size batch dim case

Reply:
addressed